internal/pkg/policy/policy_output.go

```diff
@@ -79,7 +79,7 @@ func (p *PolicyOutput) Prepare(ctx context.Context, zlog zerolog.Logger, bulker
 				return err
 			}
 
-			if ok := setMapObj(outputMap, outputAPIKey, p.Name, "api_key"); !ok {
+			if ok := setMapObj(outputMap, outputAPIKey.Agent(), p.Name, "api_key"); !ok {
 				return ErrFailInjectAPIKey
```

Comment:
I am looking at the implementation of the `#Prepare` function, looking at the signature it seems that it should be possible to inject the related dependencies to do the operation. Looking at the problems of using the `object` vs  `.Agent()`  a unit test of the behavior would have caught that. 

This seems to be a must to start adding tests that ensures the behavior of that method since this is a critical path of the Server.

Note this doesn't remove the need to have a stricter checks on the structure of the agent policy.

Reply:
There are some tests already in internal/pkg/policy/policy_output_test.go - the issue wasn't caught because the tests were expecting the incorrect value.